src/util/json_stream.cpp

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************\
+
+Module:
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+#include "json_stream.h"
+
+#include <ostream>
+
+/// Output the element delimiter to the output stream
+void json_streamt::output_delimiter()
+{
+  if(!first)
+    out << ',';
+  else
+    first = false;
+  out << '\n';
+  out << std::string((indent + 1) * 2, ' ');
+}
+
+/// Construct a new JSON array stream
+/// \param out: output stream
+/// \param indent: indentation level
+json_stream_arrayt::json_stream_arrayt(std::ostream &out, unsigned indent)
+  : json_streamt(out, indent)
+{
+  out << '[';
+}
+
+/// Output the non-streaming JSON objects and closes the current child stream
+void json_stream_arrayt::output_child_stream()
+{
+  if(!object.empty())
+  {
+    output_delimiter();
+    object["array_element"].output_rec(out, indent + 1);
+    object.clear();
+  }
```

Comment:
Noob question: why does this look quite different from `json_stream_objectt::output_child_stream`?